homeassistant/components/shelly/utils.py

```diff
@@ -6,7 +6,7 @@
 
 from aioshelly.block_device import BLOCK_VALUE_UNIT, COAP, Block, BlockDevice
 from aioshelly.const import MODEL_NAMES
-from aioshelly.rpc_device import RpcDevice
+from aioshelly.rpc_device import RpcDevice, WsServer
```

Comment:
It should not be necessary for Shelly to spin up their own WS Server. Instead, Shelly should be able to accept WebSocket connections mounted on the HA HTTP server. That way there are no extra port settings necessary.

Reply:
This would require a refactor of `WsServer` class to take a `request` in the constructor.

Code could look like this on the HA side. That way you can be available on the same port.

```python
async def get(self, request) -> WebSocketResponse:
    shellyWs = WsServer(request)
    return await WsServer.handle()
```

Reply:
actually, this code might just actually work 🤣